### PR TITLE
CHAT-176 and CHAT-174 fixes and updates to demographics table

### PR DESCRIPTION
**Pull request by roninsightrx**
- fixes issue where the table contained NA categories for sex, ethnicity and race (only when running from database, not xpt files)
- updated function to be more flexible and allow specification of demographics of interest

Examples of use [here](https://github.com/InsightRX/apollo_llm_example_data/blob/master/R/demographics_table.R)

### CODE REVIEW

File: R/create_demographics_table.R [hunk 1]
```diff
@@ -2,73 +2,122 @@
...
+  }
+  if(length(vs_demographics) == 0) {
+    demo <- data$dm
+  } else {
+    if(length(dm_demographics) == 0) {
+      demo <- vitals
+    } else {
+      demo <- data$dm %>%
+        stats::setNames(tolower(names(.))) %>%
+        merge(vitals)
```
Comment (karawoo) [inline R/create_demographics_table.R:63]: I am seeing different counts when I run the function for variables that are in `dm` only vs. when I run it for variables that are split across `dm` and `vs`. I think the reason is that we have some subject IDs in `dm` that aren't in `vs`, and at this line they are getting dropped.

Compare the counts for ethnicity in these two calls with our test data; I would expect them to be the same but they're not.

```
create_demographics_table(
  data,
  demographics = c("ethnic")
)

create_demographics_table(
  data,
  demographics = c("height", "ethnic")
)
```

**Review by karawoo — CHANGES_REQUESTED** [review]

File: R/create_demographics_table.R (hunk 1 above)
Comment (roninsightrx) [reply R/create_demographics_table.R]: good catch. Turned into left join.

File: R/create_demographics_table.R [hunk 2]
```diff
@@ -2,73 +2,122 @@
...
-    dplyr::select(-min, -max) %>%
-    tidyr::pivot_longer(cols = c(mean, sd, median, min_max)) %>%
-    stats::setNames(c("Demographic", "Statistic", "Value"))
-  
-  cat_data <- lapply(categorical, function(x) {
-    demo %>%
+  cont_data <- NULL
+  if(length(continuous) > 0) {
+    cont_data <- demo %>%
+      dplyr::select(continuous) %>%
```
[the remark this answers is not in the document]
Comment (roninsightrx) [reply R/create_demographics_table.R:82]: 👍🏻 moved the "ungroup" call a few lines up.

**Review by karawoo — APPROVED** [review]
LGTM for our test data, but do we need to anticipate cases where we have subjects in `vs` that aren't in `dm`? Maybe we should consider a full join?

**Pull request merged**